qiskit/opflow/evolutions/trotterizations/qdrift.py

```diff
@@ -73,7 +73,7 @@ def convert(self, operator: OperatorBase) -> OperatorBase:
         # and multiplication by a constant factor.
         scaled_ops = [(op * (factor / op.coeff)).exp_i() for op in operator_iter]
         sampled_ops = algorithm_globals.random.choice(
-            scaled_ops, size=(int(N * self.reps),), p=weights / lambd
+            scaled_ops, size=(int(N * self.reps),), p=np.array(weights / lambd, dtype=float)
```

Comment:
How come this is necessary?  Numpy would do this automatically if `weights/lambd` isn't an array or if it's of `dtype` `int` or something.  If the `dtype` _may_ be complex, then this will emit a warning, and the surrounding code presumably has a bug.